core/drivers/crypto/se050/core/storage.c

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <crypto/crypto.h>
+#include <se050.h>
+#include <string.h>
+
+/*
+ * pkcs-11 key deletion support:
+ * This function scans a buffer looking for a persistent key; if the key is
+ * found it calls the SE050 for its deletion
+ */
+void crypto_storage_obj_del(uint8_t *data, size_t len)
```

Comment:
That seems ad-hoc and fragile. It would be more appropriate and generic to pass an object handle, or some opaque data pointer (storage calling into crypto at object creation time, obtains an opaque context pointer, and passes the same in subsequent operations including delete).

Reply:
yes I agree with you with it being inefficient: however when the TA requests the deletion of an object, the buffer might contain not only the keys but a lot of other data ....it seemed less restrictive to allow each crypto driver to decide how better handle what to do with their marked buffers. but I know is not great...

Reply:
so, to be clear, in my tests with sks, the object that gets passed to syscall_storage_obj_del when requesting the keys tto be deleted, unfortunately is of type TEE_TYPE_DATA ...so I need to scan the buffer. maybe in the future it could receive a KEY_PAIR sort of type?

Reply:
@jforissier if you think this is going to be an issue, we can modify the interface and just return the keyid to be deleted from the SE to crypto. I can work it out in the TA and retrieve the attribute in sks

Reply:
of course, getting a pointer would be great and keeping the pointer in the session and providing at deletion would be ideal. however I dont see how that is possible with the current API. 
could you- or the storage and sks maintainer- provide some suggestions on how to do that?